### PR TITLE
Add typescript types

### PR DESCRIPTION
**Pull request by chapterjason**
Hope this is enough.

Tested with yarn link:
Screenshot 2020-12-11 at 9 37 21 PM

Fixes: #7

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { Application } from "stimulus";
```
Comment (tgalopin) [inline types.d.ts:1]: Could you use single quotes? That's the coding styles used on Symfony and UX.

That makes me think we should enable prettier on this repo :) .

**Review by tgalopin — COMMENTED** [review]
Did you test this locally?

**Comment by chapterjason** [comment]
@tgalopin Yes I tested this with yarn link and with yarn pack.

**Comment by tgalopin** [comment]
Wouldn't it make sense to provide a type package for stimulus-bridge that provide the types.d.ts file and include the webpack types?

**Comment by chapterjason** [comment]
@tgalopin You are right, not thought about it at all.

**Comment by tgalopin** [comment]
I think it's not worth creating an additional GitHub repo for this. What do you think of creating a `types` directory in this repository with a package.json inside and the types declaration? Would it make sense?

@weaverryan do you have an opinion about this?

**Comment by chapterjason** [comment]
@tgalopin sounds good, also thought about some type tests. https://github.com/Microsoft/dtslint

**Comment by chapterjason** [comment]
I took a look into the Idea with a separate directory and testing.

Testing with dtslint too much overhead for this simple function that I would like to leave it out. A *contributing* documentation would be nice to mention that when changes are made the typing must be checked and adjusted.

For the dependency of the `@types/webpack-env` I took a look into the [documentation \(> dependencies)](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies). 

>
> Here, our package depends on the `browserify` and `typescript` packages. `browserify` does not bundle its declaration files with its npm packages, so we needed to depend on `@types/browserify` for its declarations. `typescript`, on the other hand, packages its declaration files, so there was no need for any additional dependencies.
>

Based on the documentation it would be okay to add the dependency, it has no disadvantages for non TypeScript users.

**Comment by tgalopin** [comment]
It has the disadvantage of including Webpack types on every machine without any use for non TS users. IMO the node_modules dir is already big enough, we should avoid adding direct dependencies for people who won't use it.

That's why I thought a `types` directory could be better: it would allow us to have a different module for both the bridge and Webpack types, allowing people to install it easily only when they need it. WDYT?

**Comment by chapterjason** [comment]
@tgalopin Any suggestion for the package name?

**Comment by tgalopin** [comment]
`@symfony-types/stimulus-bridge` ?

File: types/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@symfony-types/stimulus-bridge",
```
Comment (tgalopin) [inline types/package.json:2]: Could you use 4 spaces?

File: types/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "Types for the Symfony Stimulus integration bridge",
+  "version": "1.1.0",
```
Comment (tgalopin) [inline types/package.json:4]: Is this version supposed to be sync with the bridge? Not sure it's useful and it's going to be forgotten for sure :) . I suggest we stick with 1.0

File: types/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "license": "MIT",
+  "main": "",
```
Comment (tgalopin) [inline types/package.json:6]: Could we remove this?

File: types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare module "@symfony/stimulus-bridge" {
+
+    import { Application } from "stimulus";
```
Comment (tgalopin) [inline types/index.d.ts:12]: Shouldn't the import be at the root level? (not sure, I don't work much with TypeScript types)

**Comment by tgalopin** [comment]
Thanks @chapterjason.

**Comment by matteodem** [comment]
Any hints when this will be deployed?

**Comment by nspyke** [comment]
For simplicity, I think this package should be under the `@types` namespace to ease autoloading.
e.g. `@types/symfony-stimulus-bridge` seems more appropriate

**Pull request merged**